Assets/Scripts/TankController.cs

```diff
@@ -5,118 +5,110 @@
 public class TankController : MonoBehaviour
 {
     public float speed;
-	public int type;
+    public int maxHealth;
+    public int type;
+    public uint reward;
     public GameObject bullet;
-	public GameObject healthBarPrefab;
-	public GameObject explosion;
-	private HealthBarController barController;
-    private Rigidbody2D rb2d;
+    public GameObject healthBarPrefab;
+    public GameObject explosion;
+    public GameObject plansza;
     public GameObject target = null;
     public bool isGunPositioned = false;
-	public int health;
-	private int maxHealth;
-	private int bulletCounter = 0;
-	private bool isQuitting = false;
-	public GameObject plansza;
-	public uint reward;
+    private int health;
+    private float healthBar;
+    private float speed_;
+    private int bulletCounter = 0;
+    private bool isQuitting = false;
+
+	private Transform tankTower;
+    private HealthBarController barController;
+    private Rigidbody2D rb2d;
+    private AudioManager audioManager;
+
+
+    void OnEnable()
+    {
+        GameObject bar = Instantiate(healthBarPrefab, this.transform);
+        bar.transform.localPosition = this.tag == "Ally" ? new Vector3(-0.7f, 0) : new Vector3(0.7f, 0);
+        barController = bar.GetComponent<HealthBarController>();
+    }
 
-	private AudioManager audioManager;
+    // Start is called before the first frame update
+    void Start()
+    {
+        rb2d = GetComponent<Rigidbody2D>();
+		tankTower = transform.GetChild(0);
+        healthBar = health = maxHealth;
+        speed_ = (this.tag == "Ally") ? speed : -speed;
+        StartAI();
+    }
 
-	// Start is called before the first frame update
-	void Start()
+    void StartAI()
     {
-		maxHealth = health;
-		GameObject bar = Instantiate(healthBarPrefab, this.transform);
-		bar.transform.localPosition = this.tag == "Ally" ? new Vector3(-0.7f, 0) : new Vector3(0.7f, 0);
-		barController = bar.GetComponent<HealthBarController>();
-		
-	}
-	private void Awake()
-	{
-		audioManager = FindObjectOfType<AudioManager>();
-	}
-
-	// Update is called once per frame
-	void Update()
+        Move();
+        if (type == 0)
+            InvokeRepeating("Fire", 0.5f, 0.8f);
+        else if (type == 1)
+            InvokeRepeating("Fire1", 0.5f, 0.4f);
+        else if (type == 2)
+            InvokeRepeating("Fire2", 0.5f, 0.2f);
+        plansza = GameObject.Find("plansza");
+    }
+    void Awake()
     {
-		if (barController)
-		{
-			Debug.Log("cyyk");
-			barController.SetSize(health / (float)maxHealth);
-		}
-
-		if (health <= 0)
-		{
-			audioManager.Play("TankExplosion");
-			Destroy(this.gameObject);
-			return;
-		}
-        if (isGunPositioned == false)
+        audioManager = FindObjectOfType<AudioManager>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    void LateUpdate()
+    {
+        healthBar = Mathf.Lerp(healthBar, health, Time.deltaTime * 2);;
+        barController.SetSize(healthBar / (float)maxHealth);
```

Comment:
I think it would be good to exclude those lines to new method
```suggestion
        updateHealthBar();
```